pkg/cni/k8sclient/k8sclient_test.go

```diff
@@ -0,0 +1,87 @@
+    client-key-data: LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlFcFFJQkFBS0NBUUVBdTIxZDRXb2RsbFNrZWFwNWVIajJ6SGhlRW5TT2hsVW1XSmp2dlZrTHZHekR0emZBCisxWjZvT3EyS3VXQW1xV2NuOEhGMW5BTlpHQnJ4V3Zydzh1T28wWGRyaVhHQ1dRZEYvbFE5dFF3SFNVVmtTU0MKN0hTRXBQeFdRUWNTc2hmclVDblhKNWRWWVpYSWR0emdyVEc1VGVERDREZ2QrT3VCYXJjYktPenphc3B6VmpSNApBK2ZSbUZDVXZxVVJFNlI5NlFiY1V0Y3NLTm9Cd2lOK3JrTTQxcjhCclFwRGQzbFBoUVlHUHZjd05HWjQvMGlCCkV0M0s3WHdFQjZ4VjhYODVCOGhWRG9XSmFFZXQ2KytqVFBIVCt3Zlc0aFd6L2hWdktvR2p3YmJXOUhFTzZ5WGIKK1hFRmQvSjhHUml5a0ZldVZhMlZaMkRza0lFR05SWllKR2JhNlFJREFRQUJBb0lCQVFDNXRLQnJZNEFUWHFSawo1SE5jMmNYM1RHQXhPd21vc2grL2N6RS9WRVpibVF6R3NkaGxjQWNxLzhRbGRxWlpGRFMrMmZrVEFvMVNZbk1GCjltdHc0WFRmODY2MUNUcFNQYjdSN1RhNmlmMXVNenRPUjNYdG9YbDRHNG0rL3FvWWliaWZBZ1hyZFhla3JBc24KTXF6dkRqQ0RxR3VMOHk1SFg5a29DbXMvdnA5a3M2WTlla3czdHNIMVNHZE5iNml2MU81bjV4OHRjNXQ3QUUyMgpzVHdEZThlZXBjZzV2MHhzRm9iU1dtNmo2a3ZCcnp2b2l4bmZ1ZWdMQ1FvcDMza1lKTUthSE95eFlZNW5YZnY5ClhNeHNPWldwcU9RU09ZVkdRZTdpUGc4aDYvOWxSZHYvbCtlTWd3MUtmaUlocENWNTdDbVMvd0pFY1lDSW1reFcKMXMweCs4OGRBb0dCQU9xMGtOU0lhSmRDaVJqaXZqWC9rSTY3K3lTN1ZGYStRbm5EN2ZvNHJGZ05nMzJGTWZmNgpQTGRvQThYRkJyYy9iZyttTGlFM3dldzBYR2lxZTlmdklVQXlnZ0FsNnlvdzdINnMwanZrcEd2UElaMzNyaHBOCmlCNzFmYVRZaldWYjgzWkk3YXVEeG1Mb1N3bStlbXJ5bWxlbXIwWlphZmI2QnN2VjVqSFkwV1ZyQW9HQkFNeHUKcjNxVmVEbzZmUTJKRDJrcUtDWmwybzRZaE9wZkxmUnJ3OFMxdFozWXAxRmRFcnVDMXN3NGRYRG9TcHJpcEdTQgo5a200VUdkSXg4ZjdCV1BhV0Nod0tXMCtiWnhZOHFoL2JKZUhRQTN2MEp4eTlaZUp1UEtBMnhtR3FoTkN3USsvClo4NGFWNkxuNnp6aFRqT0FjdStEb091TDV0akl3R3c1RlkzUzJRSDdBb0dBWFIzUVRCSG1kUVIzd2dETGVEN3gKaWo5NFQwVm5HNWNXWnByZVFxVFRjNGZCQUQ2azZYNUZNbnE0N0hEVHprWURFNEJaMHVIOU5RbzlFMlY0QnQ2ZgpzWW1ZWEJpdktTa0oydVFUOEtFd3Zua0tIRk1VcHVqVnRYcXVJNFdxNjJqRXVjd0xSejNicW9nQXBWZ0YxNEp5Cnk0MmRBbXNkQ0ZoLzg4VGtOQ2lTUXdzQ2dZRUFvQTU5QjEzbFhybVNWVG1kUGpwS2F4M055dmo1ZjhKN2FXWEMKUmMzNjN4WFVra2hydFRIUVdONVBYTklTTDBnSmE4T3cvN0QyQ3BlYUMwSEd5NUlVK2J3dlF4L2drOHUwV1NaQwo4RFJ0ZXp2cXVjTHI4L1JaUUV5UXZtQ0g0a0tlZzJUWnNpMC90Z1VjVVhNWlZndFljWncrTG8wL2RUVmdLcHRhCis4bzhLMmNDZ1lFQXk4dC9sQkZwRzNNdjBBaTlIT2VnUEhlSGZOekd1a0RpbVpmWEdqWW1qMEJOYVZjRkZya2gKaTZ6SzhPbml6bWlFLzI2bm45RG5vV3NDbm9vVVRjMnpuWlhJVCtKYXhGNnVDcG51M2JKaTVrUzBreFZYcWpmUQpWb2ZFeW9QQkdmYWVRSkhpNytYemh2UFVocDFUQkNMbGlLOUxmZkkrV0ZEU0poazZIMytWVFJBPQotLS0tLUVORCBSU0EgUFJJVkFURSBLRVktLS0tLQo=
+`
+)
+
+var _ = Describe("k8sclient tests", func() {
+	var (
+		tmpDir                string
+		validKubeconfigPath   string
+		invalidKubeconfigPath string
+	)
+
+	BeforeEach(func() {
+		var err error
+		tmpDir := GinkgoT().TempDir()
+
+		validKubeconfigPath = filepath.Join(tmpDir, "kube.conf")
+		invalidKubeconfigPath = filepath.Join(tmpDir, "kube-invalid.conf")
+
+		err = os.WriteFile(validKubeconfigPath, []byte(kubeconfigTestData), 0744)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = os.WriteFile(invalidKubeconfigPath, []byte("This is an invalid kubeconfig content"), 0744)
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
you can use DeferCleanup to cleanup the files

Reply:
i use TempDir() above , which cleans up automatically.

from ginkgo docs:


>(here we're using GinkgoT().TempDir() to access Ginkgo's implementation of t.TempDir() which cleans up after itself - there's no magic here. You could have simply called os.MkdirTemp and cleaned up afterwards yourself.)

Reply:
nice I wasn't aware of it